Formula/mage.rb

```diff
@@ -0,0 +1,25 @@
+class Mage < Formula
+  desc "Make/rake-like build tool using Go"
+  homepage "https://magefile.org"
+  url "https://github.com/magefile/mage/archive/v2.2.0.tar.gz"
+  sha256 "9a706d8b351bb395d3baca7ee98856a03f46a33ae3d884281c5841a9b8a4a924"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dst = buildpath/"src/github.com/magefile/mage"
+    src = Dir[buildpath/"*", buildpath/".*"].reject { |f| f.end_with? "." }.uniq
+
+    mkdir_p dst
+    src.each { |p| mv p, dst }
+    Dir.chdir dst do
+      system "go", "run", "./bootstrap.go"
```

Comment:
We'll either need to do this ourselves or build from a git checkout, because upstream is trying to embed the build commit & tag into the version information, which will fail from an archive tarball.

Reply:
Which is your preference?  Do you have any links that explain homebrew conventions around building from a git checkout?

I'm happy to implement either option.